packages/babel-parser/src/parser/expression.js

```diff
@@ -1153,20 +1156,27 @@ export default class ExpressionParser extends LValParser {
         if (this.state.inPipeline) {
           node = this.startNode();
 
-          if (
-            this.getPluginOption("pipelineOperator", "proposal") !== "smart"
-          ) {
-            this.raise(node.start, Errors.PrimaryTopicRequiresSmartPipeline);
-          }
+          const proposal = this.getPluginOption("pipelineOperator", "proposal");
+          const proposalToNodeType = {
+            hack: "TopicReference",
+            smart: "PipelinePrimaryTopicReference",
+          };
```

Comment:
In Babel 8 we should always use `TopicReference` for both the proposals.

Reply:
My plan had been to remove the smart-mix option in the next major version. Is there a particular reason to keep it?

Reply:
Oh if the smart proposal is "abandoned" yes, we can just remove it.